src/main/java/com/brandon3055/draconicevolution/common/tileentities/TileDislocatorInhibitor.java

```diff
@@ -0,0 +1,116 @@
+package com.brandon3055.draconicevolution.common.tileentities;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.Packet;
+import net.minecraft.network.play.server.S35PacketUpdateTileEntity;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.AxisAlignedBB;
+import net.minecraft.world.World;
+
+public class TileDislocatorInhibitor extends TileEntity {
+
+    public static final int MAXIMUM_RANGE = 16;
+    public static final int MINIMUM_RANGE = 1;
+    public static final HashMap<World, HashSet<TileDislocatorInhibitor>> inhibitors = new HashMap<>();
+
+    private boolean registered = false;
+    private int range = 5;
+
+    public static boolean isInInhibitorRange(World world, double x, double y, double z) {
+        HashSet<TileDislocatorInhibitor> list = inhibitors.get(world);
+        if (list == null) {
+            return false;
+        }
+        for (TileDislocatorInhibitor inhibitor : list) {
+            if (inhibitor.isInRange(x, y, z)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void updateEntity() {
+        if (!registered) {
+            HashSet<TileDislocatorInhibitor> map = inhibitors.computeIfAbsent(this.worldObj, k -> new HashSet<>());
+            map.add(this);
+            registered = true;
+        }
+    }
+
+    public void unregister() {
+        if (inhibitors.containsKey(this.worldObj)) {
+            HashSet<TileDislocatorInhibitor> list = inhibitors.get(this.worldObj);
+            list.remove(this);
+            if (list.isEmpty()) {
+                inhibitors.remove(this.worldObj);
+            }
+        }
+    }
+
+    public int getRange() {
+        return range;
+    }
+
+    public void setRange(int value) {
+        if (value > MAXIMUM_RANGE) {
+            value = MINIMUM_RANGE;
+        }
+        if (value < MINIMUM_RANGE) {
+            value = MAXIMUM_RANGE;
+        }
+        this.range = value;
+    }
+
+    public void increaseRange() {
+        if (range < MAXIMUM_RANGE) {
+            range++;
+        }
+    }
+
+    public void decreaseRange() {
+        if (range > MINIMUM_RANGE) {
+            range--;
+        }
+    }
+
+    public boolean isInRange(double x, double y, double z) {
+        AxisAlignedBB aabb = AxisAlignedBB.getBoundingBox(
+                this.xCoord + 0.5,
+                this.yCoord + 0.5,
+                this.zCoord + 0.5,
+                this.xCoord + 0.5,
+                this.yCoord + 0.5,
+                this.zCoord + 0.5).expand(range + 0.5, range + 0.5, range + 0.5);
+        return x >= aabb.minX && x <= aabb.maxX && y >= aabb.minY && y <= aabb.maxY && z >= aabb.minZ && z <= aabb.maxZ;
```

Comment:
This could create a bit of gc overhead, since it would create 2 AABBs per magnet per item trying to be collected. Looking aat what AABB does underneath, this could be done just with simple math instead, which I think would be favorable